Test failure TestMultiStoreEventFeed

The following test appears to have failed:

[#2045](https://circleci.com/gh/cockroachdb/cockroach/2045):

```
I0502 02:05:42.876390     446 multiraft.go:451] node 100000001: group 1 got message 12884901891->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.879448     446 multiraft.go:451] node 100000001: group 2 got message 12884901891->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.880609     446 multiraft.go:451] node 100000001: group 1 got message 8589934594->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.881146     446 multiraft.go:451] node 100000001: group 2 got message 8589934594->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.881426     446 multiraft.go:451] node 100000001: group 1 got message 8589934594->4294967297 MsgAppResp Term:6 Log:0/48
--- FAIL: TestMultiStoreEventFeed (1.99s)
	client_event_test.go:272: event feed did not match expected value. Actual values have been printed to compare with above expectation.
	client_event_test.go:273: Event feed information:
		proto.StoreID(1): []string{
				"StartStore",
				"BeginScanRanges",
				"AddRange rid=1, live=348",
				"EndScanRanges",
				"SplitRange origId=1, newId=2, origLive=854, newLive=42, origVal=715, newVal=27",
				"SplitRange origId=2, newId=3, origLive=42, newLive=0, origVal=27, newVal=0",
				"MergeRange rid=2, subId=3, live=42, subLive=0",
--
I0502 02:05:43.279984     446 multiraft.go:705] Committed Entry[3]: 6/42 EntryNormal 00000000000000003417ec1cf83bd2de: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0502 02:05:43.280065     446 multiraft.go:694] node 100000001: group 2 raft ready
I0502 02:05:43.280125     446 multiraft.go:699] HardState updated: {Term:6 Vote:4294967297 Commit:16 XXX_unrecognized:[]}
I0502 02:05:43.280938     446 multiraft.go:702] New Entry[0]: 6/16 EntryNormal 00000000000000000d89050194abc0f1: raft_id:2 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0502 02:05:43.281688     446 multiraft.go:705] Committed Entry[0]: 6/16 EntryNormal 00000000000000000d89050194abc0f1: raft_id:2 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
panic: test timed out after 5m0s

goroutine 3557 [running]:
testing.func·008()
	/usr/src/go/src/testing/testing.go:681 +0x168
created by time.goFunc
	/usr/src/go/src/time/sleep.go:129 +0x59

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0x133fa00, 0x1821cc0, 0xa3, 0xa3, 0x0)
	/usr/src/go/src/testing/testing.go:556 +0xdc3
--
	/go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:443 +0x238b
github.com/cockroachdb/cockroach/util.func·005()
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:73 +0x80
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:74 +0x133
FAIL	github.com/cockroachdb/cockroach/storage	300.035s
=== RUN TestBatchBasics
I0502 02:06:43.014191     535 rocksdb.go:88] opening in-memory rocksdb instance
I0502 02:06:43.015338     535 rocksdb.go:113] closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN TestBatchGet
I0502 02:06:43.019584     535 rocksdb.go:88] opening in-memory rocksdb instance
I0502 02:06:43.019981     535 rocksdb.go:113] closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN TestBatchMerge
I0502 02:06:43.023827     535 rocksdb.go:88] opening in-memory rocksdb instance
I0502 02:05:42.876390     446 multiraft.go:451] node 100000001: group 1 got message 12884901891->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.879448     446 multiraft.go:451] node 100000001: group 2 got message 12884901891->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.880609     446 multiraft.go:451] node 100000001: group 1 got message 8589934594->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.881146     446 multiraft.go:451] node 100000001: group 2 got message 8589934594->4294967297 MsgHeartbeatResp Term:6 Log:0/0
I0502 02:05:42.881426     446 multiraft.go:451] node 100000001: group 1 got message 8589934594->4294967297 MsgAppResp Term:6 Log:0/48
--- FAIL: TestMultiStoreEventFeed (1.99s)
	client_event_test.go:272: event feed did not match expected value. Actual values have been printed to compare with above expectation.
	client_event_test.go:273: Event feed information:
		proto.StoreID(1): []string{
				"StartStore",
				"BeginScanRanges",
				"AddRange rid=1, live=348",
				"EndScanRanges",
				"SplitRange origId=1, newId=2, origLive=854, newLive=42, origVal=715, newVal=27",
				"SplitRange origId=2, newId=3, origLive=42, newLive=0, origVal=27, newVal=0",
				"MergeRange rid=2, subId=3, live=42, subLive=0",
--
I0502 02:05:43.279984     446 multiraft.go:705] Committed Entry[3]: 6/42 EntryNormal 00000000000000003417ec1cf83bd2de: raft_id:1 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0502 02:05:43.280065     446 multiraft.go:694] node 100000001: group 2 raft ready
I0502 02:05:43.280125     446 multiraft.go:699] HardState updated: {Term:6 Vote:4294967297 Commit:16 XXX_unrecognized:[]}
I0502 02:05:43.280938     446 multiraft.go:702] New Entry[0]: 6/16 EntryNormal 00000000000000000d89050194abc0f1: raft_id:2 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
I0502 02:05:43.281688     446 multiraft.go:705] Committed Entry[0]: 6/16 EntryNormal 00000000000000000d89050194abc0f1: raft_id:2 origin_node_id:4294967297 cmd:<internal_resolve_intent:<header:<timestamp:<wall_time:0 logical:52 > cmd_id:<wall_time:0 random:0 > key:"\00
panic: test timed out after 5m0s

goroutine 3557 [running]:
testing.func·008()
	/usr/src/go/src/testing/testing.go:681 +0x168
created by time.goFunc
	/usr/src/go/src/time/sleep.go:129 +0x59

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0x133fa00, 0x1821cc0, 0xa3, 0xa3, 0x0)
	/usr/src/go/src/testing/testing.go:556 +0xdc3
--
	/go/src/github.com/cockroachdb/cockroach/multiraft/multiraft.go:443 +0x238b
github.com/cockroachdb/cockroach/util.func·005()
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:73 +0x80
created by github.com/cockroachdb/cockroach/util.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stopper.go:74 +0x133
FAIL	github.com/cockroachdb/cockroach/storage	300.035s
=== RUN TestBatchBasics
I0502 02:06:43.014191     535 rocksdb.go:88] opening in-memory rocksdb instance
I0502 02:06:43.015338     535 rocksdb.go:113] closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN TestBatchGet
I0502 02:06:43.019584     535 rocksdb.go:88] opening in-memory rocksdb instance
I0502 02:06:43.019981     535 rocksdb.go:113] closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN TestBatchMerge
I0502 02:06:43.023827     535 rocksdb.go:88] opening in-memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/client_event_test.go
+++ b/storage/client_event_test.go
@@ -73,15 +73,14 @@ func (ser *storeEventReader) recordEvent(event interface{}) {
 			event.Desc.RaftID, event.Stats.LiveBytes)
 	case *storage.SplitRangeEvent:
 		sid = event.StoreID
-		eventStr = fmt.Sprintf("SplitRange origId=%d, newId=%d, origLive=%d, newLive=%d, origVal=%d, newVal=%d",
+		eventStr = fmt.Sprintf("SplitRange origId=%d, newId=%d, origKey=%d, newKey=%d",
 			event.Original.Desc.RaftID, event.New.Desc.RaftID,
-			event.Original.Stats.LiveBytes, event.New.Stats.LiveBytes,
-			event.Original.Stats.ValBytes, event.New.Stats.ValBytes)
+			event.Original.Stats.KeyBytes, event.New.Stats.KeyBytes)
 	case *storage.MergeRangeEvent:
 		sid = event.StoreID
-		eventStr = fmt.Sprintf("MergeRange rid=%d, subId=%d, live=%d, subLive=%d",
+		eventStr = fmt.Sprintf("MergeRange rid=%d, subId=%d, key=%d, subKey=%d",
 			event.Merged.Desc.RaftID, event.Removed.Desc.RaftID,
-			event.Merged.Stats.LiveBytes, event.Removed.Stats.LiveBytes)
+			event.Merged.Stats.KeyBytes, event.Removed.Stats.KeyBytes)
 	case *storage.BeginScanRangesEvent:
 		sid = event.StoreID
 		eventStr = "BeginScanRanges"
@@ -132,8 +131,6 @@ func (ser *storeEventReader) updateCountString() string {
 // recieved by a single event reader.
 func TestMultiStoreEventFeed(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	engine.EnableMVCCComputeStatsDiagnostics(true)
-	defer engine.EnableMVCCComputeStatsDiagnostics(false)
 
 	// Create a multiTestContext which publishes all store events to the given
 	// feed.
@@ -249,25 +246,25 @@ func TestMultiStoreEventFeed(t *testing.T) {
 			"BeginScanRanges",
 			"AddRange rid=1, live=348",
 			"EndScanRanges",
-			"SplitRange origId=1, newId=2, origLive=858, newLive=42, origVal=719, newVal=27",
-			"SplitRange origId=2, newId=3, origLive=42, newLive=0, origVal=27, newVal=0",
-			"MergeRange rid=2, subId=3, live=42, subLive=0",
+			"SplitRange origId=1, newId=2, origKey=242, newKey=15",
+			"SplitRange origId=2, newId=3, origKey=15, newKey=0",
+			"MergeRange rid=2, subId=3, key=15, subKey=0",
 		},
 		proto.StoreID(2): []string{
 			"StartStore",
 			"BeginScanRanges",
 			"EndScanRanges",
-			"SplitRange origId=1, newId=2, origLive=858, newLive=42, origVal=719, newVal=27",
-			"SplitRange origId=2, newId=3, origLive=42, newLive=0, origVal=27, newVal=0",
-			"MergeRange rid=2, subId=3, live=42, subLive=0",
+			"SplitRange origId=1, newId=2, origKey=242, newKey=15",
+			"SplitRange origId=2, newId=3, origKey=15, newKey=0",
+			"MergeRange rid=2, subId=3, key=15, subKey=0",
 		},
 		proto.StoreID(3): []string{
 			"StartStore",
 			"BeginScanRanges",
 			"EndScanRanges",
-			"SplitRange origId=1, newId=2, origLive=858, newLive=42, origVal=719, newVal=27",
-			"SplitRange origId=2, newId=3, origLive=42, newLive=0, origVal=27, newVal=0",
-			"MergeRange rid=2, subId=3, live=42, subLive=0",
+			"SplitRange origId=1, newId=2, origKey=242, newKey=15",
+			"SplitRange origId=2, newId=3, origKey=15, newKey=0",
+			"MergeRange rid=2, subId=3, key=15, subKey=0",
 		},
 	}
 	if a, e := ser.perStoreFeeds, expected; !reflect.DeepEqual(a, e) {

--- a/storage/engine/mvcc.go
+++ b/storage/engine/mvcc.go
@@ -23,7 +23,6 @@ import (
 	"fmt"
 	"math"
 	"sync"
-	"sync/atomic"
 
 	"github.com/cockroachdb/cockroach/proto"
 	"github.com/cockroachdb/cockroach/util"
@@ -1355,22 +1354,6 @@ func MVCCFindSplitKey(engine Engine, raftID int64, key, endKey proto.Key) (proto
 	return humanKey, nil
 }
 
-var statsDiagnosticsEnabled int32
-
-func areStatsDiagnosticsEnabled() bool {
-	return atomic.LoadInt32(&statsDiagnosticsEnabled) == 1
-}
-
-// EnableMVCCComputeStatsDiagnostics turns stats diagnostics on or off.
-func EnableMVCCComputeStatsDiagnostics(enabled bool) {
-	oldValue := int32(0)
-	newValue := int32(1)
-	if !enabled {
-		oldValue, newValue = newValue, oldValue
-	}
-	atomic.CompareAndSwapInt32(&statsDiagnosticsEnabled, oldValue, newValue)
-}
-
 // MVCCComputeStats scans the underlying engine from start to end keys
 // and computes stats counters based on the values. This method is
 // used after a range is split to recompute stats for each
@@ -1389,10 +1372,9 @@ func MVCCComputeStats(engine Engine, key, endKey proto.Key, nowNanos int64) (pro
 	ms := proto.MVCCStats{LastUpdateNanos: nowNanos}
 	first := false
 	meta := &proto.MVCCMetadata{}
-	diag := areStatsDiagnosticsEnabled()
 
 	err := engine.Iterate(encStartKey, encEndKey, func(kv proto.RawKeyValue) (bool, error) {
-		key, ts, isValue := MVCCDecodeKey(kv.Key)
+		_, ts, isValue := MVCCDecodeKey(kv.Key)
 		if !isValue {
 			totalBytes := int64(len(kv.Value)) + int64(len(kv.Key))
 			first = true
@@ -1411,10 +1393,6 @@ func MVCCComputeStats(engine Engine, key, endKey proto.Key, nowNanos int64) (pro
 			ms.KeyCount++
 			if meta.IsInline() {
 				ms.ValCount++
-			}
-			if diag {
-				log.Infof("diag: %s meta  k=%d v=%d", key, len(kv.Key), len(kv.Value))
-				log.Infof("meta: %+v, encoded=%q", meta, kv.Value)
 			}
 		} else {
 			totalBytes := int64(len(kv.Value)) + mvccVersionTimestampSize
@@ -1444,9 +1422,6 @@ func MVCCComputeStats(engine Engine, key, endKey proto.Key, nowNanos int64) (pro
 			ms.KeyBytes += mvccVersionTimestampSize
 			ms.ValBytes += int64(len(kv.Value))
 			ms.ValCount++
-			if diag {
-				log.Infof("diag: %s value k=%d v=%d @%s", key, mvccVersionTimestampSize, len(kv.Value), ts)
-			}
 		}
 		return false, nil
 	})

